modules/grader/v2_util.go

```diff
@@ -0,0 +1,98 @@
+package grader
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"math"
+
+	"github.com/pegnet/pegnet/modules/opr"
+)
+
+// V2Payout is the amount of Pegtoshi given to the OPR with the specified index
+func V2Payout(index int) int64 {
+	if index >= 25 || index < 0 {
+		return 0
+	}
+	return 200 * 1e8
+}
+
+// ValidateV2 validates the provided data using the specified parameters
+func ValidateV2(entryhash []byte, extids [][]byte, height int32, winners []string, content []byte) (*GradingOPR, error) {
+	if len(entryhash) != 32 {
+		return nil, NewValidateError("invalid entry hash length")
+	}
+
+	if len(extids) != 3 {
+		return nil, NewValidateError("invalid extid count")
+	}
+
+	if len(extids[1]) != 8 {
+		return nil, NewValidateError("self reported difficulty must be 8 bytes")
+	}
+
+	if len(extids[2]) != 1 || extids[2][0] != 2 {
+		return nil, NewValidateError("invalid version")
+	}
+
+	o, err := opr.ParseV2Content(content)
+	if err != nil {
+		return nil, NewDecodeError(err.Error())
+	}
+
+	if o.Height != height {
+		return nil, NewValidateError("invalid height")
+	}
+
+	// verify assets
+	if len(o.Assets) != len(opr.V2Assets) {
+		return nil, NewValidateError("invalid assets")
+	}
+	for i, val := range o.Assets {
+		if i > 0 && val == 0 {
+			return nil, NewValidateError("assets must be greater than 0")
+		}
+	}
+
+	if len(o.Winners) != 10 && len(o.Winners) != 25 {
+		return nil, NewValidateError("must have exactly 10 or 25 previous winning shorthashes")
+	}
+
+	if !verifyWinnerFormat(o.GetPreviousWinners(), 10) && !verifyWinnerFormat(o.GetPreviousWinners(), 25) {
+		return nil, NewValidateError("incorrect amount of previous winners")
+	}
+
+	if !verifyWinners(o.GetPreviousWinners(), winners) {
+		return nil, NewValidateError("incorrect set of previous winners")
+	}
+
```

Comment:
We should also validate the minerid and the factoid address are valid in the v2 opr validate